src/main/java/gwt/jsonix/marshallers/xjc/plugin/builders/MainJsBuilder.java

```diff
@@ -134,16 +139,20 @@ protected static void populateJCodeModel(JCodeModel toPopulate, JClass container
         addMarshall(toPopulate, mainJsClass, containerRef, callbackMap.get(MARSHALL_CALLBACK));
     }
 
-    protected static JDefinedClass getMainJsClass(final JCodeModel toPopulate, String basePackage) throws JClassAlreadyExistsException {
-        String fullMainJsName = basePackage + "." + MAIN_JS;
+    protected static JDefinedClass getMainJsClass(final JCodeModel toPopulate, String basePackage, String mainJsName) throws JClassAlreadyExistsException {
+        String fullMainJsName = basePackage + "." + mainJsName;
         JDefinedClass jDefinedClass = toPopulate._getClass(fullMainJsName);
-
         if (jDefinedClass == null) {
-            jDefinedClass = toPopulate._class(basePackage + "." + MAIN_JS);
+            jDefinedClass = toPopulate._class(basePackage + "." + mainJsName);
             JDocComment comment = jDefinedClass.javadoc();
             String commentString = "JSInterop adapter to use for marshalling/unmarshalling.";
             comment.append(commentString);
-            jDefinedClass.annotate(toPopulate.ref(JsType.class)).param("isNative", true).param("namespace", toPopulate.ref(JsPackage.class).staticRef("GLOBAL"));
+            final JAnnotationUse classAnnotation = jDefinedClass.annotate(toPopulate.ref(JsType.class))
+                    .param("isNative", true)
+                    .param("namespace", toPopulate.ref(JsPackage.class).staticRef("GLOBAL"));
+            if (!MAIN_JS.equals(mainJsName)) {
```

Comment:
Could we please use the code below?
```java
if (!Objects.equals(MAIN_JS, mainJsName)) {
```

Reply:
@jomarko 
Is there any technical reason? Do you think we may hit a situation where a  `public static final String` may be `null` ?
I personally think that code readability depends a lot on avoiding unneeded invocation/constructs, that are only confusing (first question a careful reader could have is "why considering potentially `null` a `static final` object?")

Reply:
is `mainJsName` static final?

Reply:
@jomarko 
nope, but if the first element is not `null`, it returns `false` when the second is `null` - it is not true the other way around, though:
`public static void main(String[] args) {
        String notNull ="NOT_NULL";
        String nullString = null;
        System.out.println(notNull.equals(nullString));
        try {
            System.out.println(nullString.equals(notNull));
        } catch (NullPointerException e) {
            // expected
            System.out.println("Expected " + e.getClass().getSimpleName());
        }
        System.out.println(Objects.equals(notNull, nullString));
        System.out.println(Objects.equals(nullString, notNull));
    }`

This is Object.equals implementation (as reference)
`  public static boolean equals(Object a, Object b) {
        return (a == b) || (a != null && a.equals(b));
    }`
Basically, `Objects.equals` is a safeguard against potential null value on both/either terms, *but* if it is sure that one of them is never (and must never be) `null`, there is no need for it.
As a side-note, avoiding use of `Objects.equals` also *enforce* the requirement that the first term must never be nulled.

Reply:
@gitgabrio ok thank you, I see your point that your original code is safe, however I do not agree that using `Objects.equals` what distributed with standard java would make less readable. Approving, however for the record, my preference is `Objects.equals`.